app-backend/db/src/main/scala/com/azavea/rf/database/AnalysisDao.scala

```diff
@@ -17,47 +17,52 @@ import java.util.UUID
 import scala.concurrent.Future
 
 
-object ToolRunDao extends Dao[ToolRun] {
+object AnalysisDao extends Dao[Analysis] {
 
-  val tableName = "tool_runs"
+  val tableName = "analyses"
 
   val selectF = sql"""
     SELECT
       id, name, created_at, created_by, modified_at, modified_by, owner, visibility,
-      organization_id, execution_parameters
+      organization_id, execution_parameters, readonly
     FROM
   """ ++ tableF
 
-  def insertToolRun(newRun: ToolRun.Create, user: User): ConnectionIO[ToolRun] = {
+  def insertAnalysis(newRun: Analysis.Create, user: User): ConnectionIO[Analysis] = {
     val now = new Timestamp(new java.util.Date().getTime())
     val id = UUID.randomUUID()
 
     sql"""
-          INSERT INTO tool_runs
+          INSERT INTO analyses
             (id, name, created_at, created_by, modified_at, modified_by, owner, visibility, organization_id,
-             execution_parameters)
+             execution_parameters, readonly)
           VALUES
-            (${id}, ${newRun.name}, ${now}, ${user.id}, ${now}, ${user.id}, ${newRun.owner.getOrElse(user.id)}, ${newRun.visibility}, ${newRun.organizationId},
-             ${newRun.executionParameters})
-       """.update.withUniqueGeneratedKeys[ToolRun](
+            (${id}, ${newRun.name}, ${now}, ${user.id}, ${now}, ${user.id}, ${newRun.owner.getOrElse(user.id)},
+             ${newRun.visibility}, ${newRun.organizationId}, ${newRun.executionParameters}, ${newRun.readonly.getOrElse(false)})
+       """.update.withUniqueGeneratedKeys[Analysis](
       "id", "name", "created_at", "created_by", "modified_at", "modified_by", "owner", "visibility",
-      "organization_id", "execution_parameters"
+      "organization_id", "execution_parameters", "readonly"
     )
   }
 
-  def updateToolRun(updatedRun: ToolRun, id: UUID, user: User): ConnectionIO[Int] = {
+  def updateAnalysis(analysisUpdate: Analysis, id: UUID, user: User): ConnectionIO[Int] = {
     val now = new Timestamp(new java.util.Date().getTime())
     val idFilter = fr"id = ${id}"
 
     (sql"""
-       UPDATE tool_runs
+       UPDATE analyses
        SET
-         name = ${updatedRun.name},
+         name = ${analysisUpdate.name},
          modified_at = ${now},
          modified_by = ${user.id},
-         visibility = ${updatedRun.visibility},
-         execution_parameters = ${updatedRun.executionParameters}
-       """ ++ Fragments.whereAndOpt(ownerEditFilter(user), Some(idFilter))).update.run
+         visibility = ${analysisUpdate.visibility},
+         execution_parameters = ${analysisUpdate.executionParameters},
+         readonly = ${analysisUpdate.readonly}
+       """ ++ Fragments.whereAndOpt(ownerEditFilter(user), Some(idFilter), Some(fr"readonly = false"))).update.run
+  }
+
+  def getById(analysisId: UUID, user: User): ConnectionIO[Option[Analysis]] = {
```

Comment:
I've added a `Foo.unsafeGetbyId` that returns just a `ConnectionIO[Foo]` for use in places where I know it's fine -- you may want one here if we're going to have programmatic access in places where we're certain that the id coming in is a good one